### ISSUE TITLE
Issue with the redirection of "Learning environment" page

### ISSUE THREAD

**Issue by harridu** [kubernetes/website#36048]
The link "Learning environment" on the left panel appears to be broken. You end up somewhere in an advanced section.

**Comment by vaibhav2107** [comment]
@harridu, I think that the link is not broken but it is actually redirected to that page according to this PR #24533

**Comment by jihoon-seo** [comment]
Yeah, as @vaibhav2107 said, I can see that the link "[Learning environment](https://kubernetes.io/docs/setup/learning-environment/)" redirects to https://kubernetes.io/docs/tasks/tools/ ,
where you can find `kind` and `minikube`, the tools that let you create your own learning environment.

**Comment by jihoon-seo** [comment]
/retitle Issue with the redirection of "Learning environment" page

**Comment by sftim** [comment]
/kind feature

We could improve the user experience here so that it's less confusing for folks.
/priority backlog
/language en

**Comment by Kanika637** [comment]
I want to work on this issue, can you give me a little more context for this one @harridu

**Comment by harridu** [comment]
I just read through the documentation. For "Getting started" on https://kubernetes.io/docs/setup/ there are 3 subtopics
- Learning environment
- production environment
- best practices

I did not expect that "Learning environment" redirects me to [somewhere else in the documentation](https://kubernetes.io/docs/tasks/tools/), listing a set of tools to install, not mentioning anything about "learning kubernetes". And neither kubectl nor kubeadm are dedicated tools for a learning environment. It was just confusing.

Maybe something like "install the basic tools" would be more appropriate than "learning environment"?

https://kubernetes.io/docs/tasks/tools/ could be more precise about the suggested tools. kind and minikube **are** suitable to setup a first-steps environment, but kubectl and kubeadm are general purpose tools. And kubelet has been forgotten, as it seems.

Just my personal impression, of course.

**Comment by sftim** [comment]
Improvements are very welcome.

I'm not sure that this is a _good first issue_ - we might want folks to write a whole new page.

**Comment by sftim** [comment]
@harridu thank you for that clear explanation of your thinking.

/priority important-longterm
/remove-priority backlog

**Comment by Kanika637** [comment]
> Improvements are very welcome.
> 
> I'm not sure that this is a _good first issue_ - we might want folks to write a whole new page.

I would love to be part of the folks in writing this new page, count me in @sftim  🚀

**Comment by mehabhalodiya** [comment]
Oh, I see! I didn't know we needed a [whole new page](https://github.com/kubernetes/website/issues/36048#issuecomment-1225477888) for this issue. I am removing the `good first issue` label now.
I thought that we needed a little redirection only, so put the `good first issue` label.

/remove-good-first-issue

**Comment by Kanika637** [comment]
So @mehabhalodiya  @harridu  can I start working on it?

**Comment by harridu** [comment]
Hi Kanika,

no concerns from my side.

Sorry for the late response. Somehow I missed your EMail.


Regards

Harri

**Comment by jihoon-seo** [comment]
@Kanika637 you can self-assign yourself using this command:
/assign @Kanika637

**Comment by Kanika637** [comment]
Since I am being assigned, @harridu  can you give a little context that what exactly has to be done in this issue?

**Comment by sftim** [comment]
I think the task here is to write a new page in the Setup section, that explains what you'd need to do to set up Kubernetes so that you can learn it.

Let's explain that you need the `kubectl` client tool, so that when you configure a Kubernetes cluster, you have something that can communicate with it.

We should at a minimum explain that your options for trying out Kubernetes locally include:
- kind
- minikube

and that there are third party tools that also let you set up Kubernetes locally. Kubernetes doesn't support setting up those third party tools, but we _can_ link to a selection of vendor docs about that setup. Put kind and minikube first though.

Also let readers know about online playground environments such as https://killercoda.com/ and Play With Kubernetes. Those mean that you get to try it out without installing _anything_.

Finally, let's explain that `kubeadm` is something that someone can use to practice setting up a production like cluster. As this is not a simple task, direct readers who want advanced learning to the Production Environment page.

Maybe one day we'll have a tutorial page for setting up a local environment with `kubeadm`.

As well as writing the new page, you need to delete the existing redirect from `static/_redirects`

---

_@harridu, you might like to copy the above into the issue description if you agree with it._

**Comment by harridu** [comment]
I am not sure why I have to copy your suggestion to the description, but a new page listing several ways how to initially setup Kubernetes for the learning environment would be surely less confusing. Next to minikube and kind there might be others that could be added later.

**Comment by sftim** [comment]
> I am not sure why I have to copy your suggestion to the description.

If you do that, it helps contributors to work on the issue, and that makes it more likely that the issue gets resolved. Otherwise, people who aren't willing to read at least 21 messages and mentally assemble the final meaning will skip past this issue (they might prefer to work on an issue where the description is clear).

Overall, it's about making work easier for the volunteers who work on the documentation. I know some contributors are not volunteers and they do this as part of their role, but most _are_ volunteers. Leaving barriers in their way hinders progress.

**Comment by sftim** [comment]
Feel free to amend the description however you feel best represents the change you want made @harridu. What's important is defining the work clearly enough that someone can start work without reading through the whole issue history.

**Comment by mehabhalodiya** [comment]
@Kanika637 I don't see any updates; so unassigning you. Please feel free to assign, if you come back here again and are willing to work on 🙂 
/unassign @Kanika637

**Comment by kumarankit999** [comment]
I would like to work on this @mehabhalodiya

**Comment by kj800x** [comment]
I just wanted to add my voice here that I think this issue is important from a new user's perspective. I have attempted to learn Kubernetes a few times, and I have always been able to successfully install the tools but then am left completely confused as to what I could / should be doing next and with no awareness of what any of the concepts are.

I tried again today and realized that due to the way the docs pages work, following the pages by the order in the navigation somehow skips over the Concepts section entirely.

I start on the [Getting Started](https://kubernetes.io/docs/setup/) page. The next page in the navbar is [Learning environment](https://kubernetes.io/docs/tasks/tools/) so I click on that. I follow the instructions and set up `kubectl` and `minikube`. The next page in the sidebar is now `Administering a cluster` so I think I need to go learn about that, but all of the pages here are completely out of my depth since I have no understanding of concepts. The root cause of this confusion is the redirect when clicking on "Learning environment".

Now that I know that it exists and seems to be important to read first, I'm going to try to read the Concepts section now and hopefully will have a better time. 

I do not seem to be alone in having had this experience, I have seen [other examples online](https://www.reddit.com/r/kubernetes/comments/180mw3x/comment/ka73kzx/) where folks mention having difficulty understanding Kubernetes concepts and although I have no evidence that this redirect is the cause of the confusion, it seems probable that it's at least somewhat contributing.

**Comment by sftim** [comment]
Improvements here are welcome; I think there's between 20 and 160 hours of work to do a good job, and smaller potential improvements that take less time.

For example, can we highlight https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#what-s-next so that people use this to guide their next step?
Or can we write a page that is about setting up a local environment for learning Kubernetes?

**Comment by sftim** [comment]
If we want to do a really good job, we could host an online sandbox for learning Kubernetes. Maybe that environment could run kubectl and a shell in your local browser using something like JSLinux.

**Comment by sftim** [comment]
Would anyone like to help with this task?

**Comment by jayphan14** [comment]
Hey @sftim I would like to work on this

**Comment by lmktfy** [comment]
https://github.com/kubernetes/website/issues/51163 largely duplicates this

**Comment by lmktfy** [comment]
Given the progress to date:
/unassign @jayphan14

Kubernetes SIG Docs follows a “no [cookie licking](https://devblogs.microsoft.com/oldnewthing/20091201-00/?p=15843)” policy&mdash;so anyone can work on this issue, assigned or not.
